elliptic-curve-crate/src/coordinates.rs

```diff
@@ -0,0 +1,16 @@
+//! Generic coordinate system support
+
+use crate::ScalarBytes;
+use generic_array::ArrayLength;
+
+/// Trait for obtaining the coordinates of an affine point
+pub trait AffineCoordinates {
+    /// Size of a byte array representing an affine coordinate
+    type ScalarSize: ArrayLength<u8>;
+
+    /// x-coordinate
+    fn x(&self) -> ScalarBytes<Self::ScalarSize>;
+
+    /// y-coordinate
+    fn y(&self) -> ScalarBytes<Self::ScalarSize>;
+}
```

Comment:
This abstraction directly exposes the coordinates instead of the effect of using them. Making it a separate trait is better than having it be part of the core traits, but it still implies that coordinates are an API surface that is supported and encouraged.

I would prefer instead e.g. an `EcdsaPrimitive` trait that requires implementing the inner part of the ECDSA algorithm that requires coordinates, and then `EcdsaCurve: EcdsaPrimitive` that provides the full algorithm, which users can import without exposing the inner API unnecessarily. This is similar to the `rsa::PrivateKey: rsa::DecryptionPrimitive` split.

Separately, using `ScalarBytes` for the coordinates strongly encourages type errors during implementation of the trait. The coordinates are field elements, and scalars (of the affine point's curve) are field elements (of a different field), but the coordinates are not scalars.